web/core/components/core/charts/pie-chart/root.tsx

```
@@ -0,0 +1,73 @@
+import React, { useMemo } from "react";
+import { Cell, PieChart as CorePieChart, Pie, ResponsiveContainer, Tooltip } from "recharts";
+// plane imports
+import { TPieChartProps } from "@plane/types";
+import { Card, ECardSpacing } from "@plane/ui";
+import { cn } from "@plane/utils";
+
+export const PieChart = React.memo(<K extends string, T extends string>(props: TPieChartProps<K, T>) => {
+  const { data, dataKey, cells, className = "w-full h-96", innerRadius, outerRadius, showTooltip = true } = props;
+  // derived values
+  const itemDotClassNames = useMemo(
+    () => cells.reduce((acc, cell) => ({ ...acc, [cell.key]: cell.dotClassName }), {}),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)